packages/account/src/Sections/Profile/PersonalDetails/personal-details.tsx

```diff
@@ -0,0 +1,739 @@
+                }
+            }, 2000);
+
+            // force request to update settings cache since settings have been updated
+            const response = await WS.authorized.storage.getSettings();
+            if (response.error) {
+                setRestState({ ...rest_state, api_error: response.error.message });
+                return;
+            }
+            // Fetches the status of the account after update
+            updateAccountStatus();
+            setRestState({ ...rest_state, ...response.get_settings });
+            setIsLoading(false);
+            refreshNotifications();
+            setIsBtnLoading(false);
+            setIsSubmitSuccess(true);
+            setStartOnSubmitTimeout({
+                is_timeout_started: true,
+                timeout_callback: () => {
+                    setSubmitting(false);
+                },
+            });
+            // redirection back based on 'from' param in query string
+            const url_query_string = window.location.search;
+            const url_params = new URLSearchParams(url_query_string);
+            if (url_params.get('from')) {
+                const from = url_params.get('from') as keyof typeof routes;
+                history.push(routes[from]);
+            }
+        }
+    };
+
+    React.useEffect(() => {
+        let timeout_id: NodeJS.Timeout;
+        if (start_on_submit_timeout.is_timeout_started) {
+            timeout_id = setTimeout(() => {
+                setIsSubmitSuccess(false);
+            }, 10000);
+        }
+
+        return () => {
+            clearTimeout(timeout_id);
+        };
+    }, [start_on_submit_timeout.is_timeout_started]);
+
+    const showForm = (show_form: boolean) => setRestState({ show_form });
+
+    const isFieldDisabled = (name: string): boolean => {
+        return !!account_settings?.immutable_fields?.includes(name);
+    };
+
+    const { api_error, show_form } = rest_state;
+
+    if (api_error) return <LoadErrorMessage error_message={api_error} />;
+
+    if (is_loading || is_state_loading || !residence_list.length) {
+        return <Loading is_fullscreen={false} className='account__initial-loader' />;
+    }
+
+    const is_poa_verified = authentication_status?.document_status === 'verified';
+    const is_poi_verified = authentication_status?.identity_status === 'verified';
+
+    const is_account_verified = is_poa_verified && is_poi_verified;
+
+    //Generate Redirection Link to user based on verifiction status
+    const getRedirectionLink = () => {
+        if (!is_poi_verified) {
+            return '/account/proof-of-identity';
+        } else if (!is_poa_verified) {
+            return '/account/proof-of-address';
+        }
+        return undefined;
+    };
+
+    const PersonalDetailSchema = getPersonalDetailsValidationSchema(is_eu);
+
+    const initialValues = getPersonalDetailsInitialValues(account_settings, residence_list, states_list);
+
+    return (
+        <Formik
```

Comment:
Hi.. Are we not using PersonalDetails form component here?

Reply:
I think previously I checked with the PersonalDetails form component but the 2 structure are different so we can't reuse it inside